dags/veda_data_pipeline/veda_promotion_pipeline.py

```diff
@@ -92,8 +96,8 @@ def transfer_assets_to_production_bucket(ti=None, payload={}):
     end = EmptyOperator(task_id="end", dag=dag)
 
     collection_grp = collection_task_group()
-    mutate_payload_task = remove_thumbnail_asset()
-    extract_from_payload = extract_discovery_items_from_payload()
+    mutate_payload_task = remove_thumbnail_asset(ti=None)
+    extract_from_payload = extract_discovery_items_from_payload(ti=None)
```

Comment:
I don't think this is needed, `ti` is an [automatically-populated ](https://airflow.apache.org/docs/apache-airflow/2.10.5/tutorial/taskflow.html#accessing-context-variables-in-decorated-tasks)`kwarg`/context variable that comes from the task decorator
```suggestion
    mutate_payload_task = remove_thumbnail_asset()
    extract_from_payload = extract_discovery_items_from_payload()
```

Reply:
I had some trouble snipping `ti=None` out of some sub tasks. I agree it _shouldn't_ be needed but I have been unable to get around it in some cases. Probably indicates something about flow but sorting that out may be a bigger project

Reply:
I figured it out! I didn't have the right version of airflow installed. After upgrading to the version we're using (2.8.4) and also installing the correct version of pluggy>=1.5.0, I was able to get the unit tests working without `ti=None` set.